File: PHP/Pakaian.php
<?php

require_once __DIR__ . '/Produk.php';

// Class turunan Produk: data pakaian
// Atribut: ukuran, bahan, warna
class Pakaian extends Produk
{
    protected string $ukuran;
    protected string $bahan;
    protected string $warna;

    // Constructor
    public function __construct(
        string $id_produk,
        string $merk,
        float $harga,
        string $ukuran,
        string $bahan,
        string $warna
    ) {
        // Memanggil constructor milik Produk
        parent::__construct($id_produk, $merk, $harga);
        $this->ukuran = $ukuran;
        $this->bahan = $bahan;
        $this->warna = $warna;
    }

    // Setter dan getter for ukuran
    public function setUkuran(string $ukuran): void
    {
        $this->ukuran = $ukuran;
    }

    public function getUkuran(): string
    {
        return $this->ukuran;
    }

    // Setter dan getter for bahan
    public function setBahan(string $bahan): void
    {
        $this->bahan = $bahan;
    }

    public function getBahan(): string
    {
        return $this->bahan;
    }

    // Setter dan getter for warna
    public function setWarna(string $warna): void
    {
        $this->warna = $warna;
    }

    public function getWarna(): string
    {
        return $this->warna;
    }

    // Destructor
    public function __destruct()
    {
    }
}

File: Java/Pakaian.java
// Class turunan Produk: data pakaian
// Atribut: ukuran, bahan, warna

public class Pakaian extends Produk {
    protected String ukuran;
    protected String bahan;
    protected String warna;

    // Constructor
    public Pakaian() {
        super();
        this.ukuran = "";
        this.bahan = "";
        this.warna = "";
    }

    public Pakaian(String idProduk, String merk, float harga,
                    String ukuran, String bahan, String warna) {
        super(idProduk, merk, harga);
        this.ukuran = ukuran;
        this.bahan = bahan;
        this.warna = warna;
    }

    // Setter dan getter
    public void setUkuran(String ukuran) {
        this.ukuran = ukuran;
    }

    public String getUkuran() {
        return ukuran;
    }

    public void setBahan(String bahan) {
        this.bahan = bahan;
    }

    public String getBahan() {
        return bahan;
    }

    public void setWarna(String warna) {
        this.warna = warna;
    }

    public String getWarna() {
        return warna;
    }

    // Destructor: Java memakai garbage collector, method ini hanya
    // disertakan agar sama dengan versi C++
    @Override
    @SuppressWarnings("removal")
    protected void finalize() throws Throwable {
        super.finalize();
    }
}
